### ISSUE TITLE
Add alternative for kvstore dependency to perform node status GC of CCNP and CNP

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#10457]
The operator depends on kvstore in order to be able to perform GC of the node status for a CNP or CCNP.

This dependency can have a better alternative as the node watcher already has a store, `k8sNodeStore`, so we can use this store as well to perform the node status GC if the operator is not running with kvstore.

If the operator is running with kvstore then the operator should still use the kvstore nodeStore.

https://github.com/cilium/cilium/blob/9779b09b6a19b461ab04c97a4cac8c6f506c3eef/operator/k8s_node.go#L173

**Comment by phongvdoan** [comment]
Hi @aanm for one of my class assignements we have to contribute to an open source project, I would like to tackle this. For tests for k8sNodeStore to make sure it runs is there anything else I need to do?

**Comment by aanm** [comment]
>For tests for k8sNodeStore to make sure it runs is there anything else I need to do?

@phongvdoan thank you for volunteering! I'm sorry but I didn't get your question, can you rephrase it?

**Comment by phongvdoan** [comment]
> > For tests for k8sNodeStore to make sure it runs is there anything else I need to do?
> 
> @phongvdoan thank you for volunteering! I'm sorry but I didn't get your question, can you rephrase it?
 Am I just removing runCNPNodeStatusGC? Or it is more to it?

**Comment by aanm** [comment]
@phongvdoan there's more to it, I've edited the PR title and description to make it clear.

**Comment by daemon1024** [comment]
Hi, I would like to work on this. Can I have this assigned to me?

**Comment by pchaigno** [comment]
@daemon1024 I assigned you. Don't hesitate to reach out on Slack (e.g., #development) if you have any question.
